### DISCUSSION TITLE
docs: addSelect

### DISCUSSION THREAD

**Discussion by ramzini22** [mikro-orm/mikro-orm#5438] [Q&A] [answered]
Good day!
I wanna add select, but i didn't find a way for this
<img width="1512" alt="Снимок экрана 2024-04-10 в 11 35 35" src="https://github.com/mikro-orm/mikro-orm/assets/111016410/e601c855-d362-4ef4-9ada-7c6df1545b15">

**Comment by B4nan** [comment, marked as answer]
`qb.addSelect` expects property names, if you want to use a raw query, you need to wrap it first, either via `raw()` or the `sql` template function (https://mikro-orm.io/docs/raw-queries).

**Reply by B4nan** [reply]
Also, you should probably use a formula property instead of this, as you need to define some property anyway if you want to map the value. 

https://mikro-orm.io/docs/defining-entities#formulas

**Reply by ramzini22** [reply]
It helped me, thank u!

**Reply by ramzini22** [reply]
I added formula to my entity
<img width="646" alt="Снимок экрана 2024-04-10 в 12 22 24" src="https://github.com/mikro-orm/mikro-orm/assets/111016410/34fb8cb4-9e86-406b-9f25-6ac7b5c03e87">
but in dont use at queryBuilder
<img width="1512" alt="Снимок экрана 2024-04-10 в 12 23 39" src="https://github.com/mikro-orm/mikro-orm/assets/111016410/a00e2fd9-664f-4f8e-8bbe-bdd0d1efc787">

**Reply by B4nan** [reply]
drop the property decorator, you are not supposed to combine them. also its a bad idea to specify the alias inside your formula statically, use the callback form to have this dynamic, so it can actually work, not just when you explicitly use QB and this particular alias.

if that wont help, provide a complete repro, i cant help with screenshots, i don't even know what version you are using.

**Reply by ramzini22** [reply]
it helped me
but because i use static value at the decorator
<img width="1512" alt="Снимок экрана 2024-04-10 в 12 31 32" src="https://github.com/mikro-orm/mikro-orm/assets/111016410/1e5181a2-0e9a-4f7a-ab74-948f06d7204b">
How can i use "id" param at @Formula
<img width="1512" alt="Снимок экрана 2024-04-10 в 12 33 38" src="https://github.com/mikro-orm/mikro-orm/assets/111016410/c65b9a46-66e6-416b-b78c-bea9f67e6155">

**Reply by ramzini22** [reply]
"use the callback form to have this dynamic, so it can actually work, not just when you explicitly use QB and this particular alias."
i didnt catch it)

**Reply by B4nan** [reply]
maybe look at the docs? its the second example https://mikro-orm.io/docs/defining-entities#formulas

**Reply by ramzini22** [reply]
Good man, I also asked about "id" param from my QueryBuilder for a @Formula decorator.
Sorry, I didn't find a way how can I set parameters for all queryBuilder at the docs.
"its the second example https://mikro-orm.io/docs/defining-entities#formulas" change my sql code for posts table, but this column is formula for likes table. And i didnt find how can i set param "id" to decorator @Formula from QB
